src/libexpr/eval-cache.cc

```diff
@@ -635,7 +625,7 @@ bool AttrCursor::isDerivation()
 
 StorePath AttrCursor::forceDerivation()
 {
-    auto aDrvPath = getAttr(root->state.sDrvPath, true);
+    auto aDrvPath = getAttr("drvPath", true);
     auto drvPath = root->state.store->parseStorePath(aDrvPath->getString());
```

Comment:
Do I read correctly that this disables the former optimization of turning common strings into symbols ?

Reply:
in a manner of speaking. the optimization doesn't have a great impact here since the symbol is soon turned into a string again and used in an sqlite query, so the extra symbol table lookup we may have to do isn't very expensive in comparison.

Reply:
Right, I guess the few places where you had to move back from symbols to strings are the ones where you do not have access to a symbol table ? I see hot paths like the one in pirmops.cc still use the state->sFooBar "global" symbols.

Reply:
we do have access to the symbol in this location too (through `root->state`), in this case it just seemed more readable to not use it. wouldn't be a problem to use it here as well though if it bothers anyone.

Reply:
Just trying to understand the code really, so not bothered by anything here. From the diff it looks like a de-optimisation, but it is indeed restricted to a few places. All fine for me.